src/Provider/AbstractProvider.php

```diff
@@ -449,6 +449,7 @@ protected function getAccessTokenOptions(array $params)
 
         if ($this->getAccessTokenMethod() === self::METHOD_POST) {
             $options['body'] = $this->getAccessTokenBody($params);
+            $options['headers'] = ['content-type', 'application/x-www-form-urlencoded'];
```

Comment:
Upon reading [the spec](http://tools.ietf.org/html/rfc6749#section-4.1.3) again it is clear that this is correct content type for _both_ GET and POST requests. This is the correct solution, it just needs to be outside of the `if` statement so that it applies to any request.